packages/hint/src/lib/cli/analyze.ts

```diff
@@ -77,6 +78,57 @@ or set the flag --tracking=on|off`;
     printFrame(message);
 };
 
+const getHintsForTelemetry = (hints?: HintsConfigObject | (string | any)[]) => {
+    if (!hints) {
+        return null;
+    }
+
+    const normalizedHints = normalizeHints(hints);
+    const result = {} as HintsConfigObject;
+
+    for (const [hintId, severity] of Object.entries(normalizedHints)) {
+        result[hintId] = typeof severity === 'string' ? severity : (severity as [HintSeverity, any])[0];
+    }
+
+    return result;
+};
+
+const getHintsFromConfiguration = (configName: string, parentConfigs: string[] = []) => {
+    try {
+        const configuration = loadConfiguration(configName, parentConfigs);
+
+        return { ...getHintsFromConfigurations(configuration.extends, [configName, ...parentConfigs]), ...getHintsForTelemetry(configuration.hints) }; // eslint-disable-line no-use-before-define,@typescript-eslint/no-use-before-define
```

Comment:
To improve readability a touch:
```suggestion
        return {
            ...getHintsFromConfigurations(configuration.extends, [configName, ...parentConfigs]), // eslint-disable-line no-use-before-define,@typescript-eslint/no-use-before-define
            ...getHintsForTelemetry(configuration.hints)
        };
```